doc/rfc/5910-consistent-sql-lua-types.md

```diff
@@ -0,0 +1,1708 @@
+
+<img src="media/image5.png" />
+
+*NB! The list of supported functions is debatable.*
+
+#### Tarantool DATETIME – plan of actions
+
+1.  Benchmark possible 3rd party datetime implementations (icu-date
+    against others). Pick fastest
+    [\#5947](https://github.com/tarantool/tarantool/issues/5947).  
+    Implement Lua module wrapping his functionality;
+
+2.  Introduce date/time types serialization to Tarantool MessagePack
+    [\#5946](https://github.com/tarantool/tarantool/issues/5946);
+
+3.  Temporarily restore SQLite functions (for current TPC-H benchmark
+    facilities). Immediately mark them deprecated;
+
+4.  Introduce standard SQL date/time/timestamp types to the parser (not
+    literal yet),
+    rework the above functions to use new underlying storage instead of
+    former SQLite storage types;
+
+5.  Add selected list of required built-ins (as close to standard as
+    possible).
+    Implement them using icu-date or another library (selected at item
+    [\#1](https://github.com/tarantool/tarantool/issues/1));
+
+6.  Add datetime literals and their expressions support (date, time,
+    timestamp, intervals);
+
+## UUIDs
+
+UUID type is just a binary type of given length (128-bit). Storage,
+collation and comparison should be defined in the box implementation.
+From the SQL perspective we introduce a new primary type UUID, which
+behaves very much like any other BLOB-type.
+
+At the moment there is no plan to introduce any special handling of
+string literals with the form `'33bb9554-c616-42e6-a9c6-88d3bba4221c‘`.
+
+String literals and string expressions may be implicitly converted to
+UUID if they have proper format. Converting UUID to STRING is only
+possible via explicit cast.
+
+<img src="media/image6.png" />
+
+ISO SQL does not require any UUID support from SQL vendors, thus all
+known vendors implement it slightly differently.
+
+The worst uuid implementation, out of all those mentioned above in the
+table, is Oracle SYS\_GUID() which generates a UUID that is not *even
+RFC 4122-compliant*!
+
+Both MySQL and SQLite use similar approaches – providing a set of
+functions for conversion of string to blob, and the reverse.
+
+The most complete implementation seemingly is in PostgreSQL, which
+allows various relaxed formats for string literals which may be accepted
+as UUID – the plan is to be as close as possible to PostgreSQL here.
+Also, PostgreSQL provides several flavors of `uuid_generate*` functions
+to allow generation of different versions of standard GUIDs. Tarantool
+box generates only one version - which is frequently a limitation in
+practice (e.g. for importing 3rd party data).
+
+### Tarantool UUID – plan of actions
+
+The suggestion is:
+
+-   Introduce `UUID([version#])` functions which would allow generating any particular type of GUID;
+
+-   Add `UUID_STR(bin_uuid)` for conversion to string from binary uuid;
+
+-   Add `UUID_BLOB(string_uuid)` for explicit conversion from string to uuid blob type;
```

Comment:
I suggest to drop UUID_STR() and UUID_BLOB() from this plan. I see no need of these functions, since all conversions could be done with CAST(... AS UUID). Cast from UUID to string varbinary also can be done using CAST().